src/Mod/Test/TestPerf.py

```diff
@@ -0,0 +1,64 @@
+# ADD COPYRIGHT HERE
+
+import os
+import sys
+import unittest
+import FreeCAD as App
+import Part
+
+try:
+    from guppy import hpy
+
+    Memtest = True
+except:
+    Memtest = False
+
+try:
+    import cProfile
+
+    Pyprofile = True
+except:
+    Pyprofile = False
+
+
+class PerfTestCase(unittest.TestCase):
+    """
+    Test Case to test python syntax of all python files in FreeCAD
+    """
+
+    def setUp(self):
+        start = -1
+        for index in range(0, 20):
+            if sys.argv[index] == "--pass":
+                start = index + 1
+                break
+        print("Setup", index, sys.argv)
+        if start != -1:
+            self.fileList = sys.argv[start:]
+        if Part.Shape().ElementMapVersion == "":
+            self.tnp = ""
+        else:
+            self.tnp = ".tnp"
+        if Memtest:
+            self.memfile = open(self.fileList[0] + self.tnp + ".mprofile", "w")
+
+    def testAll(self):
+        if Pyprofile:
+            pr = cProfile.Profile()
+            pr.enable()
+        start = sys.activate_stack_trampoline
```

Comment:
This is going to make the script require Python 3.12 -- might want to note that someplace.